cmd/terramate/cli/run.go

```diff
@@ -385,10 +386,23 @@ func (c *cli) runAll(
 	err = sched.Run(func(run stackRun) error {
 		errs := errors.L()
 
+		failedTaskIndex := -1
+
 	tasksLoop:
-		for _, task := range run.Tasks {
+		for taskIndex, task := range run.Tasks {
 			acquireResource()
 
+			// For cloud sync, we always assume that there's a single task per stack.
+			cloudRun := stackCloudRun{Stack: run.Stack, Task: task}
+
+			select {
+			case <-cancelCtx.Done():
+				c.cloudSyncAfter(cloudRun, runResult{ExitCode: -1}, errors.E(ErrRunCanceled))
+				releaseResource()
+				continue tasksLoop
+			default:
+			}
```

Comment:
This is moved up so we don't try any outputs-sharing handling if the stack execution is already canceled.